apps/advisor/pkg/app/utils.go

```diff
@@ -183,3 +214,18 @@ func runStepsInParallel(ctx context.Context, spec *advisorv0alpha1.CheckSpec, st
 	wg.Wait()
 	return reportFailures, internalErr
 }
+
+func filterSteps(checkType resource.Object, steps []checks.Step) ([]checks.Step, error) {
+	ignoreSteps := checkType.GetAnnotations()[checks.IgnoreStepsAnnotation]
+	if ignoreSteps != "" && ignoreSteps != "1" { // 1 is the default value for the annotation
```

Comment:
IIUC the annotation will store a comma-separated list of step IDs, but will have a default value of `1`? What's the significance of that and how does it work if we wanted to ignore a single step that has an ID of `1`? 🙈

Reply:
Yes, the idea here is that any value in the annotation flags the feature as available in the frontend (this allow newer versions of the frontend skip the UI components related to the feature if older versions of the backend do not expose it). I'm using the same approach for the "retry step" feature.

The thing is that I also need to store in this case, the list of steps to ignore. Maybe reusing the same annotation is confusing (and yeah, it wouldn't allow for a step with an ID "1", even though that should not exist). Maybe I can use two different annotations, one for flagging the feature as available and other for the list. Does that sound better?

Reply:
Ah gotcha - thanks for explaining 👍 

It is a little confusing, though I'm not very familiar with k8s patterns for annotations so maybe this is normal design 😄 If you think it's a good idea to change and doesn't have any negative effects (like more annotations makes the API more complicated or slower for example) - then feel free. It's not a big issue from my side otherwise though 👍